### ISSUE TITLE
Abstract class not overloading an abstract method should be abstract itself.

### ISSUE THREAD

**Issue by pylint-bot** [PyCQA/pylint#179]
Originally reported by: **Yann Dirson (BitBucket: [ydirson](http://bitbucket.org/ydirson), GitHub: @ydirson?)**

---

W0223/abstract-method triggers here for class B:

```
#!python

class A(object):
    def m1(self):
        raise NotImplementedError()

class B(A):
    def m2(self):
        return self.m1()

class C(B):
    def m1(self):
        return 1
```

It's quite obvious to a human reader that B is abstract, but pylint pylint would need a hint. `class_is_abstract` apparently only considers a class to be abstract if it declares abstract methods by itself, and in fact, if I add a new abstract method in B, the warning goes away.

I could use an `@abstractclass` decorator to forcefully add an abstract method with a silly name, but that does not look like a very clean thing to do. Silencing W0223 around the class def would not declare the class to be abstract for other checks.

If only it were checking an `is_abstract` astroid property on the class, this could be added via a transform plugin eg. on `klass.__abstract__ = 1`, but with 1.1.0 it looks like there is no other way out than adding a dummy property ?

---
- Bitbucket: https://bitbucket.org/logilab/pylint/issue/179

**Comment by sam-s** [comment]
Also discussed on SO: [Pylint for half-implemented abstract classes](https://stackoverflow.com/q/30884804/850781).
Note that even adding `__metaclass__ = abc.ABCMeta` to the semi-implemented class `class B(A)` does not solve the Pylint problem.

**Comment by svenpanne** [comment]
I'm still a little bit confused by this issue. Even after reading through all replies, I'm not sure what the recommended way to handle a hierarchy like this is:

```python
class AbstractClass(object):
    __metaclass__ = abc.ABCMeta

    @abc.abstractmethod
    def foo(self):
        """abstract method one"""

    @abc.abstractmethod
    def bar(self):
        """abstract method two"""


class LessAbstractClass(AbstractClass):
    __metaclass__ = abc.ABCMeta

    def foo(self):
        return 42
```

Conceptually, the `__metaclass__`-assignment on `LessAbstractClass` is not needed, but even with that very explicit hint `pylint`, complains that `bar` is not implemented.

What should one do? Adding annotated dummy methods just for pylint's sake can't really be a solution, this confuses readers. The only sane thing I could come up with was suppressing `abstract-method` in `LessAbstractClass`, but this is not really satisfying... :unamused:

**Comment by teuneboon** [comment]
Currently one of the only issues I need the inline  `# pylint: disable=` syntax for. I feel like it's a completely legit use-case to have a class that extends from an abstract class, but is still abstract itself even though it doesn't define any specific abstract methods. Is there a case against a class being considered abstract if it has `metaclass=ABCMeta` or `__metaclass__ = ABCMeta`?

**Comment by Yevgnen** [comment]
Any news in 2019?

**Comment by PCManticore** [comment]
@Yevgnen No one got around to fix this issue just yet. If this is a problem for you, we'd be happy to accept a patch.
